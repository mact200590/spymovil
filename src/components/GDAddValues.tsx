import React, { useState } from 'react'
import { makeStyles } from '@material-ui/styles';
import { GDInput } from './GDInput'
import {GDButton }  from './GDButton';


interface Props {
    textPlaceHolder: string;
    labelButton: string
    onClick?: (value: string) => void;
}


const GDAddValues = ({ textPlaceHolder, onClick, labelButton }: Props) => {
    const classes = useStyles();
    const [value, setValue] = useState('');
    const handleOnchange = (e: any) => {
        setValue(e.target.value)
    }
    return (
        <div className={classes.container}>
            <GDInput
                typeVariant="primary"
                margin="normal"
                fullWidth={false}
                label={textPlaceHolder}
                autoFocus
                value={value}
                onChange={handleOnchange}

            />
            <GDButton className={classes.button}
                label={labelButton}
                typeVariant="primary"
                fullWidth={false}
                type="submit"
                onClick={() => {
                    onClick && onClick(value)
                    setValue("");
                }}
            />
        </div>
    )
}
export default GDAddValues;



const useStyles = makeStyles(theme => ({
    container: {
        display: "flex",
        flexDirection: "row",
    },
    button: {
        marginTop: 25,
        maxHeight: 35,
        marginLeft: 5
    }
}));