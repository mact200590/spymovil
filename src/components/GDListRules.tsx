import { List, Paper, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import React from "react";
import { Rule } from "../page/GDConfiguration";

interface Props {
  rules: Rule[];
}

const GDListRules = ({ rules }: Props) => {
  const classes = useStyles();
  return (
    <div className={classes.mainContainer}>
      <Typography
        variant="h4"
        color={"textPrimary"}
        align="center"
        className={classes.tittle}
      >
        New Rules
      </Typography>
      <Paper style={{ maxHeight: 400, overflow: "auto", minWidth: 450 }}>
        <List>
          {rules.map(item => (
            <div className={classes.row}>
              <Typography
                variant="h6"
                color={"secondary"}
                align="center"
                className={classes.move}
              >
                {item.move}
              </Typography>
              <Typography
                variant="body1"
                color={"secondary"}
                align="center"
                className={classes.middle}
              >
                => KILLS =>
              </Typography>
              <Typography
                variant="h6"
                color={"secondary"}
                align="center"
                className={classes.kill}
              >
                {item.kills}
              </Typography>
            </div>
          ))}
        </List>
      </Paper>
    </div>
  );
};

export default GDListRules;

const useStyles = makeStyles(theme => ({
  mainContainer: {
    display: "flex",
    flexDirection: "column"
  },
  tittle: {
    paddingBottom: theme.spacing(2)
  },
  row: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "center"
  },
  move: {
    display: "flex",
    flexDirection: "column",
    paddingRight: theme.spacing(2)
  },
  middle: {
    fontSize: 25,
    paddingRight: theme.spacing(2)
  },
  kill: {
    paddingBottom: theme.spacing(2)
  }
}));
