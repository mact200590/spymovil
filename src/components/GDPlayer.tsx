import React from 'react'
import { makeStyles } from '@material-ui/styles';
import GDSelect from './GDSelect';
import { SelectType } from './GDSelect'

interface Props {
  namePlayer: string
  typeVariant: SelectType;
  options: string[];
  onChange?: ((value: string) => void) | undefined;
}

const GDPlayer = ({ namePlayer, typeVariant, options, onChange }: Props) => {
  const classes = useStyles();
  return (
    <div className={classes.container}>
      <p className={classes.textPlayer}>{namePlayer}</p>
      <GDSelect typeVariant={typeVariant} options={options} onChange={onChange as any} />
    </div>
  )
}

export default GDPlayer;

const useStyles = makeStyles(theme => ({
  container: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  textPlayer: {
    fontFamily: "sans-serif",
    fontSize: 25,
    marginRight:25
  }

}));
