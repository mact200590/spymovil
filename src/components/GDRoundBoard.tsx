import { Theme, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React, { useCallback, useEffect, useMemo, useState } from "react";
import { useUseGetAllMovementsQuery, useUseGetAllRulesQuery } from "../types";
import { GDButton } from "./GDButton";
import GDSelectLabel from "./GDSelectLabel";
import { GDSpinner } from "./GDSpinner";

interface Props {
  numberRound: number;
  players: string[];
  onResult?: (player: string[]) => void;
}

interface InfoMove {
  player: string;
  move: string;
}

const GDRoundBoard = ({ numberRound, players, onResult }: Props) => {
  const classes = useStyles();
  const [indexActive, setIndexActive] = useState(0);
  const [moveActive, setMoveActive] = useState("");
  const [infoMoves, setInfoMoves] = useState<InfoMove[]>([]);
  const { loading, error, data } = useUseGetAllMovementsQuery();
  const {
    loading: loadingRules,
    error: errorRules,
    data: dataRules
  } = useUseGetAllRulesQuery();
  const rules = useMemo(() => {
    if (!dataRules || !dataRules.rules) return [];
    return dataRules.rules.map(r => ({
      id: r.id,
      move: r.move.name,
      kill: r.kill.name
    }));
  }, [dataRules]);
  const winners = useCallback(
    (infoMoves: InfoMove[]) => {
      const winners: string[] = [];
      infoMoves.forEach((move, iM) => {
        infoMoves.forEach((kill, iK) => {
          const rulesAccepted = rules.filter(
            r => r.kill === kill.move && r.move === move.move && iM !== iK
          );
          rulesAccepted.length > 0 && winners.push(move.player);
        });
      });
      return winners;
    },
    [rules]
  );
  const onClick = useCallback(() => {
    const newInfoMoves = [
      ...infoMoves,
      {
        move: moveActive,
        player: players[indexActive]
      }
    ];
    if (indexActive < players.length - 1) {
      setInfoMoves(newInfoMoves);
      setIndexActive(indexActive + 1);
    } else {
      onResult && onResult(winners(newInfoMoves));
      setIndexActive(0);
      setInfoMoves([]);
    }
  }, [moveActive, indexActive, players, infoMoves, onResult, winners]);
  const labelButton = useMemo(() => {
    return indexActive < players.length - 1 ? "next" : "play";
  }, [indexActive, players]);
  useEffect(() => {
    setMoveActive(moveActive);
  }, [infoMoves, moveActive]);
  const disable = useMemo(() => moveActive === "", [moveActive]);

  if (loading || loadingRules) return <GDSpinner />;
  if (error || errorRules) {
    //TODO: add notification
    console.log("error", error);
  }

  return (
    <div className={classes.container}>
      <Typography variant="h4" color={"textSecondary"} align="center">
        {`ROUND: ${numberRound}`}
      </Typography>

      <div className={classes.containerSelect}>
        <Typography variant="h4" color={"textSecondary"}>
          {`Player: ${players[indexActive]}`}
        </Typography>
        <GDSelectLabel
          label={"Select Move"}
          onChange={setMoveActive}
          options={
            data && data.movements ? data.movements.map(move => move.name) : []
          }
          defaultValue={moveActive}
          typeVariant={"primary"}
        />
      </div>
      <GDButton
        disabled={disable}
        className={classes.button}
        label={labelButton}
        typeVariant="primary"
        fullWidth={false}
        type="submit"
        onClick={onClick}
      />
    </div>
  );
};

export default GDRoundBoard;

const useStyles = makeStyles((theme: Theme) => ({
  container: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center"
  },
  containerSelect: {
    display: "flex",
    flexDirection: "column",
    marginTop: theme.spacing(10)
  },
  button: {
    marginTop: theme.spacing(20)
  }
}));
