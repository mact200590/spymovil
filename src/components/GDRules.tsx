import { makeStyles } from "@material-ui/styles";
import React, { useState } from "react";
import Size from "../style/size";
import { GDButton } from "./GDButton";
import GDSelect from "./GDSelect";
import { Rule } from "../page/GDConfiguration";

export type GDAddValuesType = "primary" | "secondary";

interface Props {
  textPlaceHolder: string;
  labelButton: string;
  onClick?: (move: string, skills: string) => void;
  typeVariant?: GDAddValuesType;
  movements: string[];
}

const GDRules = ({
  textPlaceHolder,
  onClick,
  labelButton,
  typeVariant = "primary",
  movements
}: Props) => {
  const classes = useStyles();
  const [move, setMove] = useState("Scissors");
  const [kill, setKills] = useState("Scissors");

  return (
    <div className={classes.container}>
      <GDSelect
        label={"Add move"}
        typeVariant={typeVariant}
        options={movements} //data && data.movements ? data.movements.map(move => move.name) : []
        onChange={setMove as any}
        styleContainer={{ minWidth: Size.box.base }}
      />
      <GDSelect
        label={"Add kill"}
        typeVariant={typeVariant}
        options={movements}
        onChange={setKills as any}
        styleContainer={{ minWidth: Size.box.base }}
      />

      <GDButton
        className={classes.button}
        label={labelButton}
        typeVariant={typeVariant}
        fullWidth={false}
        type="submit"
        onClick={() => {
          onClick && onClick(move, kill);
        }}
      />
    </div>
  );
};
export default GDRules;

const useStyles = makeStyles(theme => ({
  container: {
    display: "flex",
    flexDirection: "row"
  },
  button: {
    marginTop: 25,
    maxHeight: 35,
    marginLeft: 5
  }
}));
