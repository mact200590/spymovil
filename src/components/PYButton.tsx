import { Button } from "@material-ui/core";
import { ButtonBaseProps } from "@material-ui/core/ButtonBase";
import { makeStyles } from "@material-ui/core/styles";
import React, { useMemo } from "react";

type ButtonType = "primary" | "secondary";

type Props = {
  label: string;
  typeVariant: ButtonType;
  fullWidth: boolean;
} & ButtonBaseProps;

export const PYButton: React.FC<Props> = ({
  label,
  typeVariant,
  fullWidth,
  ...rest
}) => {
  const classes = useStyles();
  const variant = useMemo(
    () => ({ primary: "contained", secondary: "outlined" }[typeVariant] as any),
    [typeVariant]
  );
  console.log(classes);
  return (
    <Button
      type="submit"
      {...rest}
      fullWidth
      variant={variant}
      color="primary"
      className={classes.primary}
    >
      {label}
    </Button>
  );
};

const useStyles = makeStyles((theme: any) => (
   {
    primary: {
      color: theme.palette.primary
    },
    secondary: {
      color: theme.palette.primary
    }
  }
));
