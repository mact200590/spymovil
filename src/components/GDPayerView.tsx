import React, { useState } from 'react'
import GDtext from './GDText'
import GDPlayer from './GDPlayer'
import { PYButton } from './PYButton'
import { makeStyles } from "@material-ui/core/styles";
import GDAddValues from './GDAddValues';

interface Props {
  onClick?: (player1: string, player2: string) => void;
}

const GPPlayerView = ({ onClick }: Props) => {
  const classes = useStyles();
  const [valuePlayer1, setValuePlayer1] = useState('')
  const [valuePlayer2, setValuePlayer2] = useState('')

  return (
    <div className={classes.row}>
      <div className={classes.column}>
        <GDtext title={"Enter Player's Name "} />
        <div className={classes.center}>
          <GDPlayer namePlayer={'Payer 1'} onChange={setValuePlayer1} options={['Scissors', 'Paper', 'Rock']} typeVariant={'primary'} />
          <GDPlayer namePlayer={'Payer 2'} onChange={setValuePlayer2} options={['Scissors', 'Paper', 'Rock']} typeVariant={'primary'} />

          <PYButton className={classes.mainButton}
            label="Start"
            typeVariant="primary"
            fullWidth={true}
            type="submit"
            onClick={() => {
              console.log("Value Player", valuePlayer1, valuePlayer2)
            }}
          />
        </div>
      </div>
      <div className={classes.button}>
        <GDAddValues labelButton='Add player' textPlaceHolder={"Add Player"} onClick={() => console.log("Imprime en la BD")} />
      </div>
    </div>
  )
}

export default GPPlayerView;

const useStyles = makeStyles(theme => ({
  row: {
    display: "flex",
    flexDirection: "row",
    justifyContent: 'center'
  },
  column: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  },
  center: {
    paddingLeft: 45
  },
  mainButton: {
    maxWidth: 240,
    marginLeft: 10
  },
  button: {
    paddingTop: 150,
  }
}));