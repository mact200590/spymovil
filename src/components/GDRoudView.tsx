import React, { useState } from 'react';
import GDText from './GDText';
import GDSelect from './GDSelect';
import { makeStyles } from '@material-ui/styles';
import { PYButton } from './PYButton';

interface Props {
    numberRound: number
    activePlayer: string
    options: string[]
    labelButton: string
    onClick?: (move: string) => void
}

const GDRoundView = ({ numberRound, activePlayer, options, onClick, labelButton }: Props) => {
    const classes = useStyles();
    const [value, setValue] = useState('')
    return (
        <div className={classes.column}>
            <h1>ROUND: {numberRound}</h1>
            <h2>{activePlayer}</h2>
            <div className={classes.row}>
                <p>Select Move</p>
                <GDSelect typeVariant={"primary"} options={options} onChange={setValue as any} />
            </div>
            <PYButton className={classes.button}
                label={labelButton}
                typeVariant="primary"
                fullWidth={false}
                type="submit"
                onClick={() => {
                    console.log("dsfdfsfsdf" + value)
                }}
            />
        </div>
    )
}

export default GDRoundView;

const useStyles = makeStyles(theme => ({
    column: {
        display: "flex",
        flexDirection: "column",
        justifyContent: 'center',
        paddingLeft: 10
    },
    row: {
        display: "flex",
        flexDirection: "row"
    },
    button: {
        maxWidth: 100,
        marginLeft: 15,
        marginTop: 10,
        marginBottom: 10

    }
}));