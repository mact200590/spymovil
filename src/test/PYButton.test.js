import "@testing-library/jest-dom/extend-expect";
import { cleanup, render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";
import { act, create } from "react-test-renderer";
import { PYButton } from "../components/PYButton";

afterEach(cleanup);
it("Button with all mandatory props  or other props ", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <PYButton label="Buscarte" typeVariant="primary" fullWidth={false} />,
    div
  );
});

it("It renders correctly", () => {
  const { getByTestId } = render(
    <div data-testid="button">
      <PYButton label="Buscar" typeVariant="primary" fullWidth={false} />
    </div>
  );
  expect(getByTestId("button")).toHaveTextContent("Buscar");
});

it("Matches Snapshot", () => {
  let tree;
  act(() => {
    tree = create(
      <div data-testid="button">
        <PYButton label="Buscar" typeVariant="primary" fullWidth={false} />
      </div>
    );
  });

  expect(tree.toJSON()).toMatchSnapshot();
});

it("Matches snapshot with React Testing Library", () => {
  const { container } = render(
    <PYButton label="Buscar" typeVariant="primary" fullWidth={false} />
  );
  expect(container).toMatchSnapshot();
});