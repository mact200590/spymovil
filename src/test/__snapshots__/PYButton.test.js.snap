// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  data-testid="button"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Buscar
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Matches snapshot with React Testing Library 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Buscar
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
